### DISCUSSION TITLE
Option to disable CSS transitions when switching themes

### DISCUSSION THREAD

**Discussion by andrewdoro** [mantinedev/mantine#1533] [Feature requests]
It would be great to have the option to disable CSS transitions when switching themes. We could add `disableTransitions` an option to the `ColorSchemeProvider`  and change the `useMantineColorScheme` hook to return `toggleColorScheme` as function that disables CSS transitions for a few ms before switching the themes.

https://github.com/mantinedev/mantine/blob/97349accc5632aefabb1a86d91a6f4b221e534cf/src/mantine-styles/src/theme/ColorSchemeProvider.tsx#L27-L33

https://github.com/mantinedev/mantine/blob/97349accc5632aefabb1a86d91a6f4b221e534cf/src/mantine-styles/src/theme/ColorSchemeProvider.tsx#L11-L21

## Current behaviour
In this example, the button has a transition property for the `backgroundColor` when hovering, but this also applies when the theme changes. This example is not that bad, but when we have multiple transitions that have different durations values, it will result in an incohesive result.

https://user-images.githubusercontent.com/17547035/170835483-094e6f58-5ab6-4b68-8567-274fc19b362a.mp4

## Proposed behaviour

https://user-images.githubusercontent.com/17547035/170835679-8a854228-8f39-4a36-af26-fcc1e152a939.mp4

**Comment by rtivital** [comment]
Do you have an idea how to implement that? Would you like to send a PR with this feature?

**Reply by andrewdoro** [reply]
yes, I am gonna do a PR

**Reply by andrewdoro** [reply]
added a PR #1540
